fhir-persistence/src/main/java/com/ibm/fhir/persistence/payload/PayloadPersistenceHelper.java

```diff
@@ -42,24 +42,23 @@
      * @param compress
      * @return
      */
-    public static InputOutputByteStream render(Resource resource, boolean compress) throws FHIRPersistenceException {
+    public static InputOutputByteStream render(Resource resource, boolean compress) throws FHIRGeneratorException, IOException {
         InputOutputByteStream ioStream = new InputOutputByteStream(DATA_BUFFER_INITIAL_SIZE);
         
         if (compress) {
             try (GZIPOutputStream zipStream = new GZIPOutputStream(ioStream.outputStream())) {
                 FHIRGenerator.generator(Format.JSON, false).generate(resource, zipStream);
-                zipStream.close();
             } catch (IOException | FHIRGeneratorException x) {
-                logger.log(Level.SEVERE, "Resource: '" + resource.getClass().getSimpleName() + "/" + resource.getId() + "'", x);
-                throw new FHIRPersistenceException("Store payload failed");
+                logger.log(Level.SEVERE, "Failed generating resource: '" + resource.getClass().getSimpleName() + "/" + resource.getId() + "'", x);
+                throw x;
```

Comment:
I wonder if there was a reason we omitted the cuased by in the past...

In general I don't love to log at sever level AND throw...that often leads to the same error being logged many times.

Reply:
Yeah, although because we try not to attach sensitive information to exceptions in case it's propagated back to the caller. I think it's better to log locally but then not log again where we actually catch and handle the exception.